static/src/javascripts/projects/common/modules/social/pinterest.js

```diff
@@ -0,0 +1,35 @@
+define([
+    'fastdom',
+    'bean',
+    'common/utils/$'
+], function (
+    fastdom,
+    bean,
+    $
+) {
+    function launchOverlay(event) {
+        event.preventDefault();
+
+        fastdom.read(function () {
+            $('img:not(.gu-image):not(.responsive-img)').each(function (img) {
+                fastdom.write(function () {
+                    $(img).attr('data-pin-nopin', 'true');
+                });
+            });
+        });
+
+        require(['js!https://assets.pinterest.com/js/pinmarklet.js?r=' + new Date().getTime()]);
+    }
+
+    return function () {
+        fastdom.read(function () {
```

Comment:
You don't actually have to do a `read` when just selecting from the DOM. You only need to do a `read` when you're querying some property of the DOM that will force layout (e.g., 'what are the dimensions of this div?'). Some info on things that trigger layout here: http://gent.ilcore.com/2011/03/how-not-to-trigger-layout-in-webkit.html